tests/Doctrine/Tests/ORM/Functional/Ticket/GH11149/GH11149Test.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket\GH11149;
+
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * @group GH11149
+ * @requires PHP 8.1
+ */
+class GH11149Test extends OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->setUpEntitySchema([
+            Locale::class,
+            Product::class,
+            ProductTranslation::class,
+        ]);
+    }
+
+    public function testEagerLoadAssociationIndexedByEntity(): void
+    {
+        $this->_em->persist($product = new Product(11149));
+        $this->_em->persist($locale = new Locale(LocaleCode::French));
+        $this->_em->persist(new ProductTranslation($product, $locale));
+
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $product = $this->_em->find(Product::class, 11149);
+        static::assertInstanceOf(Product::class, $product);
+        static::assertCount(1, $product->translations);
+
+        $translation = $product->translations->get(LocaleCode::French->value);
```

Comment:
Ah now we have another issue. The code inside the test file itself does not parse. Pretty sure we've fixed a similar issue before, I'll try to find the solution.

Reply:
Ah well I see you've found a solution yourself, great! Otherwise, we have another way out: https://github.com/doctrine/persistence/commit/f5844651d4f1e23f7e2e4d035c7700564a1e4787

Reply:
We can set the test file to be in a PHP 7 compatible syntax, we just need to avoid using the backed enum in this class. Luckily that was an easy fix.